vipgo-helper.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+/**
+ * Prevent the maintenance_mode plugin returning a 503 HTTP status to Nagios.
+ *
+ * Maintenance_mode sets a 503 header on page requests if maintenance_mode is enabled and this leads to  Nagios
+ * reporting lots of server errors for sites that are just in maintenance_mode. This function sets the filter
+ * response that maintenance_mode uses to determine if it shoudl set teh 503 status header or not.
+ *
+ * @return bool Should maintenance_mode set a 503 header
+ */
+function wpcom_vip_maintenance_mode_do_not_respond_503_for_nagios() {
+
+	// Check if request is coming from Nagios.
+	if ( ! empty( $_SERVER['HTTP_USER_AGENT'] ) && false === strpos( $_SERVER['HTTP_USER_AGENT'], 'check_http' ) ) {
```

Comment:
This might be highly opinionated bit of feedback, but anyway. If this function is preventing 503 for nagios request, I would expect the conditional to be evaluated as true for nagios request, not for non-nagios. Making it the other way round (true for non-nagios) makes the code harder to read. Compare to:

```
if ( ! empty( $_SERVER['HTTP_USER_AGENT'] ) && false !== strpos( $_SERVER['HTTP_USER_AGENT'], 'check_http' ) ) {
    $respond503 = false; // The request comes from Nagios so deny the 503 header being set.
}

return $respond_503;
```

It, imho, makes the code more readable and requires less comments along the way.